### PR TITLE
add dtype property to graph

### PR DESCRIPTION
**Pull request by Bartzi**
makes things easier

### CODE REVIEW

File: length/tests/data_set_tests/test_data_sets.py [hunk 1]
```diff
@@ -34,7 +34,7 @@ def test_mnist_reading_test():
     iterations = 0
     for batch in data_set.train:
         assert (100, 784) == batch.data.shape
-        assert DTYPE == batch.data.data.dtype
+        assert DTYPE == batch.data.dtype
         assert (100,) == batch.labels.shape
         assert np.uint8 == batch.labels.data.dtype
```
Comment (Jopyth) [inline length/tests/data_set_tests/test_data_sets.py]: Use it here too? :cat:

File: length/tests/data_set_tests/test_data_sets.py [hunk 2]
```diff
@@ -46,7 +46,7 @@ def test_mnist_reading_train():
     iterations = 0
     for batch in data_set.test:
         assert (100, 784) == batch.data.shape
-        assert DTYPE == batch.data.data.dtype
+        assert DTYPE == batch.data.dtype
         assert (100,) == batch.labels.shape
         assert np.uint8 == batch.labels.data.dtype
```
Comment (Jopyth) [inline length/tests/data_set_tests/test_data_sets.py]: and here?

File: length/tests/data_set_tests/test_data_sets.py [hunk 3]
```diff
@@ -58,7 +58,7 @@ def test_mnist_reading_2_dimensions():
     iterations = 0
     for batch in data_set.train:
         assert (100, 28, 28) == batch.data.shape
-        assert DTYPE == batch.data.data.dtype
+        assert DTYPE == batch.data.dtype
         assert (100,) == batch.labels.shape
         assert np.uint8 == batch.labels.data.dtype
```
Comment (Jopyth) [inline length/tests/data_set_tests/test_data_sets.py]: and here?

File: length/tests/data_set_tests/test_data_sets.py [hunk 4]
```diff
@@ -70,7 +70,7 @@ def test_mnist_reading_3_dimensions():
     iterations = 0
     for batch in data_set.train:
         assert (100, 1, 28, 28) == batch.data.shape
-        assert DTYPE == batch.data.data.dtype
+        assert DTYPE == batch.data.dtype
         assert (100,) == batch.labels.shape
         assert np.uint8 == batch.labels.data.dtype
```
Comment (Jopyth) [inline length/tests/data_set_tests/test_data_sets.py]: and here?

**Review by Jopyth — CHANGES_REQUESTED** [review]
:question:

**Comment by Bartzi** [comment]
done, please review again :smile:

**Review by Jopyth — APPROVED** [review]
:japanese_castle:

**Pull request merged**